fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceDAOImpl.java

```diff
@@ -547,7 +563,23 @@ public Resource insert(Resource resource, List<ExtractedParameterValue> paramete
             long latestTime = System.nanoTime();
             double dbCallDuration = (latestTime-dbCallStartTime)/1e6;
 
-            resource.setId(stmt.getLong(7));
+            long logicalResourceId = stmt.getLong(7);
```

Comment:
I know robin recommended to save this to a local variable, but I really don't like this variable name.  It makes it seem like it is the LOGICAL_RESOURCE_ID from this table but it is actually the RESOURCE_ID.

Aside:  I also don't love that we use the name RESOURCE_ID for this column...I'd have preferred something like `VERSIONED_RESOURCE_ID` or `RESOURCE_VERSION_ROW_ID` or similar, but we've had that since forever.
Since we probably don't want to touch the schema, maybe we can at least use that to name the variable?
```suggestion
            long versionedResourceRowId = stmt.getLong(7);
```

Reply:
Actually, scratch that, it *is* the logical resource id that we get back.  In this case the variable was named correctly but the update command won't.  We need to get the RESOURCE_ID (and not the LOGICAL_RESOURCE_ID) that was just written so that we can update just that row.